backend/src/errors.rs

```diff
@@ -3,89 +3,94 @@ use axum::{
+/// ```ignore
+/// // Given an instance of `CustomError::InternalServerError`:
+/// let error = CustomError::InternalServerError;
+/// let response = error.into_response();
+/// assert_eq!(response.status(), StatusCode::INTERNAL_SERVER_ERROR);
+/// ```
+///
+/// ```ignore
+/// // For a database unique violation:
+/// let pg_error = sqlx::Error::Database(/*... simulated unique violation error ...*/);
+/// let custom_error = CustomError::from(pg_error);
+/// let response = custom_error.into_response();
+/// assert_eq!(response.status(), StatusCode::BAD_REQUEST);
+/// ```
 impl IntoResponse for CustomError {
     fn into_response(self) -> Response {
         let (status, error_message) = match self {
-            Self::InternalServerError => (
+            CustomError::BadRequest(msg) => (StatusCode::BAD_REQUEST, msg),
+            _ => (
                 StatusCode::INTERNAL_SERVER_ERROR,
                 "Internal Server Error".to_string(),
             ),
-            Self::BadRequest => (StatusCode::BAD_REQUEST, "Bad Request".to_string()),
-            Self::NotFound(detail) => (StatusCode::NOT_FOUND, detail),
-            Self::DatabaseError(db_error) => (
-                StatusCode::INTERNAL_SERVER_ERROR,
-                format!("Database Error: {}", db_error),
-            ),
         };
+
         let body = Json(json!({
             "error": error_message,
         }));
 
         (status, body).into_response()
     }
 }
-
-#[derive(Debug)]
-pub enum SignupError {
-    NonUniqueIdError,
-    UnknownQueryError,
-    UnknownDatabaseError,
-}
-
-impl From<sqlx::Error> for SignupError {
-    fn from(error: sqlx::Error) -> Self {
-        match error {
-            sqlx::Error::Database(db_error) => {
-                let pg_error = db_error.downcast::<sqlx::postgres::PgDatabaseError>();
-                match pg_error.code() {
-                    "23505" => {
-                        println!("Duplicate user ID.");
-                        SignupError::NonUniqueIdError
-                    }
-                    _ => {
-                        println!("-- An error the server didn't account for --");
-                        println!("{:?}", pg_error);
-                        SignupError::UnknownQueryError
-                    }
-                }
-            }
-            _ => {
-                println!("Something else happened");
-                SignupError::UnknownDatabaseError
-            }
-        }
-    }
-}
-
-impl std::fmt::Display for SignupError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            SignupError::NonUniqueIdError => {
-                write!(f, "Duplicate email contained a duplicate key.")
-            }
-            SignupError::UnknownQueryError => {
-                write!(f, "Database query contained an unspecified error.")
-            }
-            SignupError::UnknownDatabaseError => write!(f, "Database error, not query related."),
-        }
-    }
-}
```

Comment:
This code felt redundant. Though we can easily add more custom error names for the struct like SignupError, EmailNotFound, etc.